light-model-client/src/commonMain/kotlin/org/modelix/client/light/LightModelClient.kt

```diff
@@ -40,6 +80,13 @@ class LightModelClient internal constructor(
     private var exceptions: MutableList<ExceptionData> = ArrayList()
     private var currentModelQuery: ModelQuery? = null
     private var unappliedQuery: ModelQuery? = null
+    private val modelQLClient: ModelQLClient? = when (connection) {
```

Comment:
Suggestion: Replace pattern matching with polymorphism.

`IConnection` could declare and `WebsocketConnection` could implement a method like `getModelQLClient`.